server/whoami.go

```diff
@@ -0,0 +1,54 @@
+package server
+
+import (
+	"net/http"
+
+	"github.com/TheYeung1/yata-server/middleware/auth"
+	log "github.com/sirupsen/logrus"
+)
+
+type WhoAmIOutput struct {
```

Comment:
I don't expect for us to keep all these fields but I figured I'd keep them for now until we settle on which ones we need.

In reality I imagine we will eventually have a proper "users" table which this API will query.